### ISSUE TITLE
Cumulative error in rotations

### ISSUE THREAD

**Issue by andrewtarzia** [lukasturcani/stk#184]
Multiplication of rotation matrices upon construction cause a decrease in building block diameter. We expect this to be due to numerical errors in the multiplication, which accumulate over time.

Normalizing matrices is the first step in solving. But a change to using quaternions and to doing all rotations in one step.

Minimum working example:
[cubes.zip](https://github.com/lukasturcani/stk/files/4845791/cubes.zip)

**Comment by andrewtarzia** [comment]
How come this is not in the stk paper project?

**Comment by lukasturcani** [comment]
I can add it, but I did not consider this priority. In that I don't think this is a big problem atm (might be wrong here), and it does not actually affect the content of the paper.

**Comment by andrewtarzia** [comment]
I think we should add it, and I do think it is important to remove bugs like this before publication.

**Comment by lukasturcani** [comment]
I cannot reproduce the error from the example. Both of the numbers printed are identical.

```
▶ python min_example.py
init  16.246521105147405
const 16.246521105147405
```

**Comment by andrewtarzia** [comment]
I also cannot reproduce this with the latest version of stk (master) - will explore further.

**Comment by lukasturcani** [comment]
Will re-open if reproducible.

**Comment by andrewtarzia** [comment]
This issue is definitely still occurring, see the attached mol file (as a text file): look at the nodes on the corner of the cubes, one is much smaller than the rest. Will need to figure out how to reproduce on small scale. We must have changed something about the building blocks ability to update that messed with the minimum example 
[TEMPdel.txt](https://github.com/lukasturcani/stk/files/5505293/TEMPdel.txt)

**Comment by lukasturcani** [comment]
Right, so maybe the building block is not changing but the problem is that when placed on a topology graph, the building black size is different? That makes much more sense, because building blocks are and were immutable. However, the modified clones may have ended up with cumulative errors?

**Comment by andrewtarzia** [comment]
Yes, I think that is the case. So the issue definitely still needs fixing.

In the old version the example did work.

**Comment by lukasturcani** [comment]
Can you still update the example please with comments that specify exactly what is wrong. Just so that its super for someone who opens it up with no context.

**Comment by andrewtarzia** [comment]
It is hard to state exactly what is wrong in the code example, I updated the ZIP below and will state the atoms to look for to visualise the effect.

The cluster bonded to the Zinc atom with id 632.

[cubes.zip](https://github.com/lukasturcani/stk/files/5509141/cubes.zip)

**Comment by lukasturcani** [comment]
Fixed by #255
